fava_investor/modules/performance/accounts_config.py

```diff
@@ -0,0 +1,41 @@
+import re
+from typing import List
+
+from core import FavaLedger
+
+from fava_investor.modules.performance.balances import ConfigDict
+
+
+class AccountsConfig:
+    def __init__(self, value: List[str], internal: List[str], external: List[str]):
+        """
+        Full lists of all accounts considered value/internal/external. useful when building queries
+        for matching account or these with joinstr(other_accounts)
+        """
+        self.value = value
+        self.internal = internal
+        self.external = external
+
+    @classmethod
+    def from_dict(cls, ledger: FavaLedger, config: ConfigDict) -> 'AccountsConfig':
+        re_inv = cls._prepare_regexp(config['value'])
+        re_int = cls._prepare_regexp(config['internal'])
+        inv = []
+        internal = []
+        external = []
+        for account, _ in ledger.accounts.items():
+            if re_inv.match(account):
+                inv.append(account)
+            elif re_int.match(account):
+                internal.append(account)
+            else:
+                external.append(account)
+
+        return AccountsConfig(inv, internal, external)
+
+    @staticmethod
+    def _prepare_regexp(inv_):
```

Comment:
Since we're doing a simple regexp or, we should instead allow users to do this in their config itself, and remove this method completely. This way, the user can specify exactly what they want without worrying about how their regexp might be changed. Users will thus specify a single regexp (not a list).

Reply:
Are you certain that regexp support is needed in config? They certainly are not very user friendly and tree of accounts is grouping them anyway. Does it ever make sense to exclude subaccount and include parent for example? or exclude parent and include it's subaccount.

Reply:
I can see the need for both cases. An example of the former: when a specific highly volatile subaccount is desired to be excluded so as to not throw off the numbers. Or, when one wants to selectively pick only subaccounts in various parts of the tree that are with one specific institution.

What concerns do you have with allowing regexps for all these three sets?